pyrtl/rtllib/pyrtlfloat/_float_utills.py

```diff
@@ -0,0 +1,104 @@
+import pyrtl
+
+from ._types import FPTypeProperties
+
+
+class FloatUtils:
+    @staticmethod
+    def get_sign(fp_prop: FPTypeProperties, wire: pyrtl.WireVector) -> pyrtl.WireVector:
+        return wire[fp_prop.num_mantissa_bits + fp_prop.num_exponent_bits]
+
+    @staticmethod
+    def get_exponent(
+        fp_prop: FPTypeProperties, wire: pyrtl.WireVector
+    ) -> pyrtl.WireVector:
+        return wire[
+            fp_prop.num_mantissa_bits : fp_prop.num_mantissa_bits
+            + fp_prop.num_exponent_bits
+        ]
+
+    @staticmethod
+    def get_mantissa(
+        fp_prop: FPTypeProperties, wire: pyrtl.WireVector
+    ) -> pyrtl.WireVector:
+        return wire[: fp_prop.num_mantissa_bits]
+
+    @staticmethod
+    def is_zero(fp_prop: FPTypeProperties, wire: pyrtl.WireVector) -> pyrtl.WireVector:
+        return (FloatUtils.get_mantissa(fp_prop, wire) == 0) & (
+            FloatUtils.get_exponent(fp_prop, wire) == 0
+        )
+
+    @staticmethod
+    def is_inf(fp_prop: FPTypeProperties, wire: pyrtl.WireVector) -> pyrtl.WireVector:
```

Comment:
It would really help to have some comments with brief definitions and references for key floating point concepts (NaN, inf, denormalized (how are these different?), guard, round, sticky, ...)